src/measure_theory/measure/measure_space.lean

```diff
@@ -2601,6 +2601,117 @@ begin
+  (As_disj : pairwise (disjoint on As)) (Union_As_finite : μ (⋃ i, As i) < ∞) :
+  set.finite {i : ι | ε ≤ μ (As i)} :=
+begin
+  by_contradiction con,
+  set substantial := {i : ι | ε ≤ μ (As i)} with h_substantial,
+  let pickseq := set.infinite.nat_embedding _ con,
+  set js := (λ n, (pickseq.to_fun n).val) with js_def,
+  have js_inj : function.injective js,
+  { intros n m hnm,
+    simp only [js_def, function.embedding.to_fun_eq_coe, subtype.val_eq_coe] at hnm,
+    have equ : pickseq.to_fun n = pickseq.to_fun m, by { ext, exact hnm, },
+    exact pickseq.injective equ, },
+  have disj : pairwise (disjoint on (λ (n : ℕ), As (js n))),
+    from λ n m hnm b hb, As_disj (js n) (js m) (function.injective.ne js_inj hnm) hb,
+  have huge : μ (⋃ n, As (js n)) = ∑' n, μ (As (js n)), from measure_Union disj (λ n, As_mble _),
+  have large_terms : ∀ n, ε ≤ μ (As (js n)), from λ n, (pickseq.to_fun n).property,
+  have diverges : ∑' n, μ (As (js n)) = ∞,
+  { have obs : ∑' (n : ℕ), ε ≤ ∑' n, μ (As (js n)), from ennreal.tsum_le_tsum large_terms,
+    rw (show ∑' (n : ℕ), ε = ∞, from ennreal.tsum_const_eq_top_of_ne_zero ε_pos.ne.symm) at obs,
+    exact eq_top_iff.mpr obs, } ,
+  have sub : (⋃ n, As (js n)) ⊆ (⋃ i, As i), from Union_subset (λ n, subset_Union _ _),
+  have shouldnot := lt_of_le_of_lt (show μ (⋃ n, As (js n)) ≤ μ (⋃ i, As i), from measure_mono sub)
+                    Union_As_finite,
+  rw [huge, diverges] at shouldnot,
+  exact lt_irrefl _  shouldnot,
+end
+
+/-- If the union of disjoint measurable sets has finite measure, then there are only
+countably many members of the union whose measure is positive. -/
+lemma countable_meas_pos_of_disjoint_Union' {ι : Type*} [measurable_space α]
+  {μ : measure α} {As : ι → set α} (As_mble : ∀ (i : ι), measurable_set (As i))
+  (As_disj : pairwise (disjoint on As)) (Union_As_finite : μ (⋃ i, As i) < ∞) :
+  set.countable {i : ι | 0 < μ (As i)} :=
+begin
+  set posmeas := {i : ι | 0 < μ (As i)} with posmeas_def,
+  set fairmeas := λ (n : ℕ) , {i : ι | μ (As i) ≥ 1/n} with fairmeas_def,
+  have countable_union : posmeas = (⋃ n, fairmeas(n)) ,
+  { have fairmeas_eq : ∀ n, fairmeas n = (λ i, μ (As i)) ⁻¹' Ici (n : ℝ≥0∞)⁻¹,
+    { intro n,
+      simpa only [fairmeas_def, one_div, ge_iff_le], },
+    simpa only [fairmeas_eq, posmeas_def, ← preimage_Union,
+                ennreal.Union_Ici_inv_nat, pos_iff_ne_zero.symm], },
+  have countable_pieces : ∀ n, set.countable (fairmeas n),
+  { intros n,
+    apply finite.countable,
+    apply finite_const_le_meas_of_disjoint_Union _ As_mble As_disj Union_As_finite,
+    simp only [one_div, ennreal.add_eq_top, ennreal.nat_ne_top, ne.def, ennreal.one_ne_top,
+               not_false_iff, ennreal.inv_pos, or_self], } ,
+  rw countable_union,
+  exact countable_Union countable_pieces,
+end
+
+/-- In a σ-finite space, among disjoint measurable sets, only countably many can have positive
+measure. -/
+lemma countable_meas_pos_of_disjoint_Union
```

Comment:
The main part of this proof should be a lemma about `tsum` on `ennreal`: if you have a tsum on ennreal that is finite, then there are countable many positive terms. I feel like we should have that already somewhere.

Reply:
I did not actually find this fact about `tsum`, but it is a similar argument as in the PR, at least if one first reformulates `summable.tendsto_cofinite_zero` as the finiteness of the set of indices where the summands exceed a given positive level.

Reply:
Actually... How did you plan to reduce this to a fact about `tsum` anyway?

If the indexing type of the disjoint union is not countable to start with, the measure of the disjoint union is not the `tsum` of measures of the sets in the union... And if the indexing type is countable, there is nothing to prove.

Reply:
Oh, that is a good point. I didn't think that through very well. Is it true that the measure of a (not necessarily countable) disjoint union is always at least the sum of the measures?

Reply:
Yes, I added `tsum_meas_le_meas_Union_of_disjoint` and use it to shorten the main proof.